cpp/src/communicator/mpi.cpp

```diff
@@ -122,33 +124,34 @@ std::unique_ptr<Communicator::Future> MPI::send(
 std::unique_ptr<Communicator::Future> MPI::send(
     std::unique_ptr<Buffer> msg, Rank rank, Tag tag
 ) {
-    if (!msg->is_ready()) {
-        logger().warn("msg is not ready. This is irrecoverable, terminating.");
-        std::terminate();
-    }
+    RAPIDSMPF_EXPECTS(msg->is_latest_write_done(), "msg must be ready");
     RAPIDSMPF_EXPECTS(
         msg->size <= std::numeric_limits<int>::max(),
         "send buffer size exceeds MPI max count"
     );
     MPI_Request req;
-    RAPIDSMPF_MPI(MPI_Isend(msg->data(), msg->size, MPI_UINT8_T, rank, tag, comm_, &req));
+    auto data = const_cast<std::byte*>(msg->data());
+    RAPIDSMPF_MPI(MPI_Isend(data, msg->size, MPI_UINT8_T, rank, tag, comm_, &req));
```

Comment:
Why not `exclusive_data_access`?

Reply:
but we don't need the `const_cast`, that was some old code. We now have:
```c++
RAPIDSMPF_MPI(MPI_Isend(msg->data(), msg->size, MPI_UINT8_T, rank, tag, comm_, &req));
```